src/main/java/org/truffleruby/core/encoding/EncodingNodes.java

```diff
@@ -656,8 +660,15 @@ protected DynamicObject encodingGetObjectEncoding(DynamicObject object) {
         }
 
         @Specialization(guards = "isRubyRegexp(object)")
-        protected DynamicObject encodingGetObjectEncodingRegexp(DynamicObject object) {
-            return getRubyEncodingNode.executeGetRubyEncoding(Layouts.REGEXP.getSource(object).getEncoding());
+        protected DynamicObject encodingGetObjectEncodingRegexp(DynamicObject object,
+                @Cached("createBinaryProfile()") ConditionProfile hasRegexpSource) {
+            final Rope regexpSource = Layouts.REGEXP.getSource(object);
+
+            if (hasRegexpSource.profile(regexpSource == null)) {
```

Comment:
Since the profile is named `hasRegexpSource` the condition should be `regexpSource != null`.
Could you adapt the condition and swap the branches?